### PR TITLE
Added !i for information to interactive telegram client

### PR DESCRIPTION
**Pull request by JosXa**

### CODE REVIEW

File: telethon_examples/interactive_telegram_client.py [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
-from telethon import TelegramClient, ConnectionMode
-from telethon.errors import SessionPasswordNeededError
-from telethon.tl.types import (
-    UpdateShortChatMessage, UpdateShortMessage, PeerChat
-)
 from telethon.utils import get_display_name
 
+from telethon import ConnectionMode, TelegramClient
+from telethon.errors import SessionPasswordNeededError
+from telethon.tl.types import (PeerChat, UpdateShortChatMessage, UpdateShortMessage)
```
Comment (Lonami) [inline telethon_examples/interactive_telegram_client.py]: Prefer:
```python
from telethon.tl.types import (
    PeerChat, UpdateShortChatMessage, UpdateShortMessage
)
```

File: telethon_examples/interactive_telegram_client.py [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
 
 def sprint(string, *args, **kwargs):
     """Safe Print (handle UnicodeEncodeErrors on some terminals)"""
     try:
         print(string, *args, **kwargs)
     except UnicodeEncodeError:
-        string = string.encode('utf-8', errors='ignore')\
-                       .decode('ascii', errors='ignore')
+        string = string.encode('utf-8', errors='ignore') \
+            .decode('ascii', errors='ignore')
```
Comment (Lonami) [inline telethon_examples/interactive_telegram_client.py]: Why did this change (or why did you commit it)?

File: telethon_examples/interactive_telegram_client.py [hunk 3]
```diff
@@ -234,8 +235,8 @@ def run(self):
 
                         # And print it to the user
                         sprint('[{}:{}] (ID={}) {}: {}'.format(
-                               msg.date.hour, msg.date.minute, msg.id, name,
-                               content))
+                            msg.date.hour, msg.date.minute, msg.id, name,
```
Comment (Lonami) [inline telethon_examples/interactive_telegram_client.py]: Was better before.

File: telethon_examples/interactive_telegram_client.py [hunk 4]
```diff
@@ -190,6 +190,7 @@ def run(self):
             print('  !d <msg-id>: Deletes a message by its id')
             print('  !dm <msg-id>: Downloads the given message Media (if any).')
             print('  !dp: Downloads the current dialog Profile picture.')
+            print('  !i: prints information about this chat..')
```
Comment (Lonami) [inline telethon_examples/interactive_telegram_client.py]: All informations start with upper case, but you didn't. (`prints` → `Prints`).

File: telethon_examples/interactive_telegram_client.py [hunk 5]
```diff
@@ -270,6 +271,23 @@ def run(self):
                     else:
                         print('No profile picture found for this user!')
 
+                elif msg == '!i':
+                    blacklist_attrs = ['from_reader', 'on_response', 'pretty_format', 'resolve', 'rpc_error',
```
Comment (Lonami) [inline telethon_examples/interactive_telegram_client.py]: This is horribly ugly. You should use `entity.to_dict()` and iterate over that, no blacklist needed.

File: telethon_examples/interactive_telegram_client.py [hunk 6]
```diff
@@ -270,6 +271,23 @@ def run(self):
...
+                                       'constructor_id', 'subclass_of_id', 'confirm_received', 'result']
+                    interesting_attributes = [
+                        x for x in dir(entity)
+                        if x.lower() not in blacklist_attrs and not (x.startswith('_') or x.startswith('__'))
+                    ]
+
+                    longest_attr = max(len(x) for x in interesting_attributes)
+
+                    for attr in interesting_attributes:
+                        print("{:<{width}} : {}".format(
```
Comment (Lonami) [inline telethon_examples/interactive_telegram_client.py]: Prefer:
```python
something.format(
    ...
)
```

Over:

```python
something.format(
    ...)
```

File: telethon_examples/interactive_telegram_client.py [hunk 7]
```diff
@@ -356,6 +374,6 @@ def update_handler(self, update):
             else:
                 who = self.get_entity(update.from_id)
                 sprint('<< {} @ {} sent "{}"'.format(
-                       get_display_name(which), get_display_name(who),
-                       update.message
+                    get_display_name(which), get_display_name(who),
```
Comment (Lonami) [inline telethon_examples/interactive_telegram_client.py]: Same, this change wasn't necessary.

File: telethon_examples/interactive_telegram_client.py (hunk 2 above)
Comment (JosXa) [reply telethon_examples/interactive_telegram_client.py]: It's autoformatting of pycharm..

Comment (JosXa) [reply telethon_examples/interactive_telegram_client.py]: we seem to have different settings

File: telethon_examples/interactive_telegram_client.py (hunk 5 above)
Comment (JosXa) [reply telethon_examples/interactive_telegram_client.py]: Well then how do we know which attributes are interessting and which ones to ignore? __Some__ list of attribs must be in there, in any form

Comment (Lonami) [reply telethon_examples/interactive_telegram_client.py]: Okay sure but using `.to_dict()` would save you from knowing library internals, like `from_reader`, `on_response`, `pretty_format`, `resolve` and `rpc_error`. Oh, look, all of the relevant ones.

**Comment by Lonami** [comment]
Rather funny I changed the core of the pull request. Oh well, it works, shows that `.to_dict()` is a thing.

**Pull request merged**